Refactor Settings UI-UX

kumarpalsinh25 opened the pull request:
This PR contains serval UI Improvements in the settings screens.

Most of the changes are in settings page and profile page. Other files changes are just due to naming changes. So out 18 files changes, there are only 2 or 3 file which have major changes other have minor things to look on :)

- - -
Video Reference
- - -
https://github.com/acterglobal/a3/assets/51526480/c7f5904a-2b26-4d2a-85e8-41e0c98993b3

- - -
Mobile
- - -
![Settings](https://github.com/acterglobal/a3/assets/51526480/986c77be-1410-4887-bef8-791befdfa805)
![Settings (1)](https://github.com/acterglobal/a3/assets/51526480/b3375afa-6205-40d6-b802-fb00ad062929)
- - -
Desktop
- - -
![Settings (2)](https://github.com/acterglobal/a3/assets/51526480/bc179a71-ea16-483f-95fa-e8c29cf9ec31)
![Settings (3)](https://github.com/acterglobal/a3/assets/51526480/97c2dd70-ca8f-46f3-9a2a-82b0f4b11b2b)

kumarpalsinh25 commented:
@gnunicorn 

> The first entry for the profile isn't really clear as a settings option IMHO.

The first entry will look like below.
![Screenshot 2024-03-07 at 7 03 05 PM](https://github.com/acterglobal/a3/assets/51526480/8b6301ac-1972-4842-ae3f-11ed84ad246b)

> How about changing the title to "My Profile" and the subtitle to show the displayName (user-id as fallback)?

Sorry but not getting you, can you explain bit more about this?

> Edit screen :  the way to change the avatar is completely unclear 

How about if we put pencil icon on bottom left corner of the avatar?

> that the display name and username have the same design despite one being change-able (display name) while the other isn't ... and then have a save-button that effectively only is about the display name (avatar is changed whenever you select a new one regardless of whether you clicked 'save') isn't really making sense ...

We can do one thing here..
1) Differentiate username field in such way that it will look no-editable
2) Make display as view only and give pencil icon at trailing side to open edit display name pop-up as it was there previously and remove save button

gtalha07 commented on app/lib/features/home/widgets/sidebar_widget.dart:
```diff
@@ -61,7 +61,7 @@ class SidebarWidget extends ConsumerWidget {
               margin: const EdgeInsets.only(top: 8),
               child: InkWell(
-                onTap: () => context.pushNamed(Routes.myProfile.name),
+                onTap: () => context.goNamed(Routes.settings.name),
```
You sure this wouldn't break the routing part ? Need to be careful thinking about changing route transition type...

gtalha07 commented on app/lib/features/profile/pages/my_profile_page.dart:
```diff
@@ -1,121 +1,42 @@
+      // close loading
+      EasyLoading.dismiss();
+
+      context.pop();
```
Where does this pop to? Back to settings menu? If that is, we don't want to navigate back. Not sure if its needed since Loader is changed and dismissed already by here.

kumarpalsinh25 replied:
Yes, it is for going back to setting menu. Open for suggestion here :)

gtalha07 commented on app/lib/features/profile/pages/my_profile_page.dart:
```diff
@@ -129,56 +50,66 @@ class MyProfilePage extends ConsumerWidget {
+  AppBar _buildAppbar(BuildContext context) {
+    return AppBar(
+      title: Text(
+        'Profile',
```
Should be `My Profile` as addressed by ben.

gnunicorn replied:
According to the design questionaire results we should speak of "You" and "Your". So "Your profile". But I think "profile" is fine enough here.

gnunicorn commented:
> How about if we put pencil icon on bottom left corner of the avatar?

Yeah, sounds good.

gtalha07 commented on app/lib/features/profile/pages/my_profile_page.dart:
```diff
@@ -129,56 +50,66 @@ class MyProfilePage extends ConsumerWidget {
+  }
+
+  Widget _buildBody(BuildContext context) {
+    return Consumer(
```
We don't need this ```Consumer``` wrapper do we? Could be simply like :
```suggestion
    Widget _buildBody(BuildContext context){
    final account = ref.watch(accountProfileProvider);
    return account.when(
    ...
    }
```

gtalha07 commented on app/lib/features/settings/widgets/settings_menu.dart:
```diff
@@ -24,216 +23,227 @@ class SettingsMenu extends ConsumerWidget {
+  Widget _settingMenuSection({
+    required BuildContext context,
+    required String sectionTitle,
+    bool isDanderZone = false,
```
minor word correction! ;)

gtalha07 commented on app/lib/features/settings/widgets/settings_menu.dart:
```diff
@@ -24,216 +23,227 @@ class SettingsMenu extends ConsumerWidget {
+            color: enable ? null : Theme.of(context).disabledColor,
+          ),
+        ),
+        subtitle: subTitle == null
```
Could be simplified to 
```suggestion
        subtitle: subTitle ?? Text(
                subTitle,
                style: titleStyles?.copyWith(
                  color: enable ? null : Theme.of(context).disabledColor,
                ),
              ),
```

kumarpalsinh25 commented on app/lib/features/settings/widgets/settings_menu.dart:
```diff
@@ -24,216 +23,227 @@ class SettingsMenu extends ConsumerWidget {
     }
 
-    final shouldGoNotNamed = isDesktop || size.width > 770;
+    final shouldGoNotNamed = isDesktop && size.width > 770;
```
Yes understand this but this is require change based on the recent changes and doesn't effecting any other places :)

gnunicorn replied:
why is it both though? why not just `size_width`-dependent? I'd really prefer to remove any instance of `isDesktop` we can get rid of ... it is almost always too generic...

kumarpalsinh25 commented:
@gnunicorn 

1) 
Added 'Edit Profile' label
![Screenshot 2024-03-07 at 8 24 21 PM](https://github.com/acterglobal/a3/assets/51526480/aa78cbcc-86fc-43ee-9eb1-d8c7e02802c8)

2) 
- Added pencil icon over the profile avatar
- Changed the color of username filed to make it like readonly
![Screenshot 2024-03-07 at 8 24 43 PM](https://github.com/acterglobal/a3/assets/51526480/1154e7c6-8b40-421d-9682-2d7a6d5284f5)

gnunicorn commented on app/lib/features/settings/pages/settings_page.dart:
```diff
@@ -0,0 +1,93 @@
+  }
+
+  Widget _userProfileUI() {
+    return Consumer(
+      builder: (context, ref, child) {
```
not necessary here either, if we made this a consumer-widget ... which I think, I'd prefer...

gnunicorn commented on app/lib/features/settings/pages/settings_page.dart:
```diff
@@ -0,0 +1,93 @@
+            );
+          },
+          error: (e, trace) => Text('error: $e'),
+          loading: () => const Text('loading'),
```
no skeletons :( ... :really_sad_panda:

kumarpalsinh25 replied:
My Bad! let me do it :)

gnunicorn commented on app/lib/features/settings/widgets/settings_menu.dart:
```diff
@@ -24,216 +23,227 @@ class SettingsMenu extends ConsumerWidget {
+              iconData: Atlas.key_monitor_thin,
+              iconColor: colorSelected(Routes.settingSessions),
+              title: 'Sessions',
+              subTitle: 'Your active device sessions',
```
```suggestion
              subTitle: 'Your active devices',
```

let's simplify this.

kumarpalsinh25 commented on app/lib/features/profile/pages/my_profile_page.dart:
```diff
@@ -129,236 +50,193 @@ class MyProfilePage extends ConsumerWidget {
+            );
+          },
+          error: (e, trace) => Text('error: $e'),
+          loading: () => const Text('loading'),
```
How can I forgot this. Bad thing :(
Let me do it :)

kumarpalsinh25 commented on app/lib/features/settings/pages/blocked_users.dart:
```diff
@@ -68,7 +68,7 @@ class BlockedUsersPage extends ConsumerWidget {
     final users = ref.watch(ignoredUsersProvider);
     return WithSidebar(
-      sidebar: const SettingsMenu(),
+      sidebar: const SettingsPage(),
```
Ahh, understood your thought here. Let me change it to `SettingsMenu`. :)

kumarpalsinh25 replied:
We already have `SettingsMenu()` class in widget directory which container all the menus expect profile option.
`SettingsPage()` represents entire settings page which container upper profile menu and below side uses `SettingsMenu` component.
